pkg/controller/controller.go

```diff
@@ -462,42 +479,50 @@ func convertCatalog(in *osb.CatalogResponse) ([]*v1alpha1.ServiceClass, error) {
 			if err != nil {
 				err = fmt.Errorf("Failed to marshal metadata\n%+v\n %v", svc.Metadata, err)
 				glog.Error(err)
-				return nil, err
+				return nil, nil, err
 			}
-			ret[i].ExternalMetadata = &runtime.RawExtension{Raw: metadata}
+			serviceClasses[i].ExternalMetadata = &runtime.RawExtension{Raw: metadata}
 		}
 
-		ret[i].SetName(svc.Name)
+		serviceClasses[i].SetName(svc.Name)
+
+		// set up the plans using the ServiceClass Name
+		plans, err := convertServicePlans(svc.Plans, svc.Name)
+		if err != nil {
+			return nil, nil, err
+		}
+		servicePlans = append(servicePlans, plans...)
 	}
-	return ret, nil
+	return serviceClasses, servicePlans, nil
 }
 
-func convertServicePlans(plans []osb.Plan) ([]v1alpha1.ServicePlan, error) {
-	ret := make([]v1alpha1.ServicePlan, len(plans))
-	for i := range plans {
-		ret[i] = v1alpha1.ServicePlan{
-			Name:        plans[i].Name,
-			ExternalID:  plans[i].ID,
-			Free:        (plans[i].Free != nil && *plans[i].Free),
-			Description: plans[i].Description,
+func convertServicePlans(plans []osb.Plan, serviceClassName string) ([]*v1alpha1.ServicePlan, error) {
+	servicePlans := make([]*v1alpha1.ServicePlan, len(plans))
+	for i, plan := range plans {
+		servicePlans[i] = &v1alpha1.ServicePlan{
+			ExternalID:      plan.ID,
+			Free:            (plan.Free != nil && *plan.Free),
+			Description:     plan.Description,
+			ServiceClassRef: apiv1.LocalObjectReference{Name: serviceClassName},
 		}
+		servicePlans[i].SetName(plan.Name + plan.ID)
```

Comment:
s/`plan.Name + plan.ID`/`plan.Name + "-" + plan.ID`/

Reply:
huh?

Reply:
add a dash. Its problematic to ask people to see/type: `wordword` instead of `word-word`. There's ugly ux and then there's crazy ux.

Reply:
it's not wordword, it's wordID